### PR TITLE
MH-13242 Set disable_search_threshold for chosen globally

### PR DESCRIPTION
**Pull request by staubesv**
Just realised that Opencast 6.x and newer come with a new default here which we don't want at our site. 

This pull request simplifies the code by settings disable_search_threshold globally which also makes it easier for Adopters to customize this value if desired.

### CODE REVIEW

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html [hunk 1]
```diff
@@ -494,8 +487,6 @@ <h2 translate="EVENTS.EVENTS.DETAILS.HEADER" translate-values="{{titleParams}}">
                                     data-width="'100%'"
                                     ng-model="workflow.id"
                                     ng-change="changeWorkflow(false)"
-                                    data-disable-search-threshold="0"
-                                    data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html:498]: Is it intentional that you're removing this as well? If so, why?

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html [hunk 2]
```diff
@@ -582,8 +573,6 @@ <h2 translate="EVENTS.EVENTS.DETAILS.HEADER" translate-values="{{titleParams}}">
                               <select chosen
                                       pre-select-from="acls"
                                       data-width="'200px'"
-                                      data-disable-search-threshold="0"
-                                      data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html:586]: Same question here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html [hunk 3]
```diff
@@ -635,8 +624,6 @@ <h2 translate="EVENTS.EVENTS.DETAILS.HEADER" translate-values="{{titleParams}}">
                                       pre-select-from="roles"
                                       ng-disabled="((tab == 'access') && !$root.userIs('ROLE_UI_EVENTS_DETAILS_ACL_EDIT'))"
                                       data-width="'360px'"
-                                      data-disable-search-threshold="0"
-                                      data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html:639]: Same question here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html [hunk 4]
```diff
@@ -141,8 +141,6 @@ <h2 translate="EVENTS.SERIES.DETAILS.HEADER" translate-values="{{titleParams}}">
                                     pre-select-from="acls"
                                     ng-disabled="(tab == 'permissions') && (!$root.userIs('ROLE_UI_SERIES_DETAILS_ACL_EDIT') || aclLocked)"
                                     data-width="'200px'"
-                                    data-disable-search-threshold="0"
-                                    data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html:145]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html [hunk 5]
```diff
@@ -193,8 +191,6 @@ <h2 translate="EVENTS.SERIES.DETAILS.HEADER" translate-values="{{titleParams}}">
                                     pre-select-from="roles"
                                     ng-disabled="(tab == 'permissions') && (!$root.userIs('ROLE_UI_SERIES_DETAILS_ACL_EDIT') || aclLocked)"
                                     data-width="'360px'"
-                                    data-disable-search-threshold="0"
-                                    data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html:197]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html [hunk 6]
```diff
@@ -54,8 +54,6 @@
                           </p>
                           <select chosen pre-select-from="wizard.step.acls"
                                          data-width="'200px'"
-                                         data-disable-search-threshold="0"
-                                         data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html:58]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html [hunk 7]
```diff
@@ -103,8 +101,6 @@
                         <td>
                           <select chosen pre-select-from="wizard.step.roles"
                                          data-width="'360px'"
-                                         data-disable-search-threshold="0"
-                                         data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html:107]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html [hunk 8]
```diff
@@ -466,8 +452,6 @@
                          tabindex="1"
                          focushere
                          ng-change="wizard.step.changeWorkflow()"
-                         data-disable-search-threshold="0"
-                         data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html:470]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html [hunk 9]
```diff
@@ -520,8 +504,6 @@
                                   data-width="'200px'"
                                   tabindex="1"
                                   focushere
-                                  data-disable-search-threshold="0"
-                                  data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html:524]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html [hunk 10]
```diff
@@ -570,8 +552,6 @@
                         <td>
                           <select chosen
                                   data-width="'360px'"
-                                  data-disable-search-threshold="0"
-                                  data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html:574]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html [hunk 11]
```diff
@@ -79,8 +79,6 @@
                           <select chosen
                                   pre-select-from="wizard.step.acls"
                                   data-width="'200px'"
-                                  data-disable-search-threshold="0"
-                                  data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html:83]: And here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html [hunk 12]
```diff
@@ -128,8 +126,6 @@
                         <td>
                           <select chosen
                                   data-width="'360px'"
-                                  data-disable-search-threshold="0"
-                                  data-search_contains="true"
```
Comment (KatrinIhler) [inline modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html:132]: And here.

**Review by KatrinIhler — COMMENTED** [review]
So I'm assuming the reasons for declining #21 don't apply anymore? Not that I'm complaining about making things more consistent.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html (hunk 1 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html:498]: data-search_contains is already set to true in the global configuration. No need to set this here again.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html (hunk 2 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html:586]: data-search_contains is set to "true" globally.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html (hunk 4 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html:145]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html (hunk 5 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/series-details.html:197]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html (hunk 6 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html:58]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html (hunk 3 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html:639]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html (hunk 7 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-acl.html:107]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html (hunk 8 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html:470]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html (hunk 9 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html:524]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html (hunk 10 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-event.html:574]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html (hunk 11 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html:83]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html (hunk 12 above)
Comment (staubesv) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/wizards/new-series.html:132]: Same here.

File: modules/admin-ui/src/main/webapp/scripts/app.js [hunk 13]
```diff
@@ -117,6 +117,7 @@ angular.module('adminNg', [
 .config(['chosenProvider', function (chosenProvider) {
   chosenProvider.setOption({
     'search_contains': true,
```
Comment (staubesv) [inline modules/admin-ui/src/main/webapp/scripts/app.js:119]: @KatrinIhler This sets search_contains for all drop down boxes globally. That is why I have removed the many "data-search_contain" attributes: They are simply redundant.

**Comment by staubesv** [comment]
> So I'm assuming the reasons for declining #21 don't apply anymore? Not that I'm complaining about making things more consistent.

They still do apply - we just react differently at our side since the situation has changed: At the time #21 was in review, PR #21 would have changed the search_threshold to 0. Meanwhile, the official Opencast codebase already sets this to zero.
The motivation for this PR is that we will not follow the community considering this change and we will add a custom patch to our codebase that reverts this. That is easier to do if we only need to adapt one line of code. Hence you see this PR from our side, this time.

File: modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html (hunk 1 above)
Comment (KatrinIhler) [reply modules/admin-ui/src/main/webapp/scripts/shared/partials/modals/event-details.html:498]: You're right, I overlooked that...

**Comment by KatrinIhler** [comment]
> > So I'm assuming the reasons for declining #21 don't apply anymore? Not that I'm complaining about making things more consistent.

> They still do apply - we just react differently at our side since the situation has changed: At the time #21 was in review, PR #21 would have changed the search_threshold to 0. Meanwhile, the official Opencast codebase already sets this to zero.

Where does that happen? It looks like at least in edit-event the threshold is still 8.

**Comment by staubesv** [comment]
@KatrinIhler It comes from here: https://github.com/opencast/opencast/pull/389/files. I've adapted edit-event since it is quite the same situation as the detail modals and wizards. Didn't adapt the filters, though.

**Review by KatrinIhler — APPROVED** [review]
@staubesv Ah, I see. Alright, let's merge this then!

**Pull request merged**